src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -184,6 +194,27 @@ public static bool IsDisposeImplementation([NotNullWhen(returnValue: true)] this
                 method.IsImplementationOfInterfaceMethod(null, iDisposable, "Dispose");
```

Comment:
```
            return method.ReturnsVoid &&
                method.Parameters.Length == 0 &&
                method.IsImplementationOfInterfaceMethod(null, iDisposable, "Dispose");
```

@Evangelink Why there is no warning reported on the use of `.Length == 0` here?
I think it should report to use .IsEmpty. There is an existing analyzer for that. Analyzer bug? or I'm missing something?

Reply:
Not sure why there is no diagnostic, I would also expect one, I will see if we have the same behavior in some test.

> Beside that point, is there anything left for the PR?

Let me do a new pass on the PR.